tf_quant_finance/black_scholes/approximations/american_option.py

```diff
@@ -0,0 +1,306 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Baron-Adesi Whaley approximation of the Black Scholes equation to calculate
+ the price of a batch of American options."""
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tf_quant_finance as tff
```

Comment:
The tests are failing. You need to import the modules explicitly because at this stage there is no library yet. That is, lilke you add 
import tf_quant_finance.black_scholes.implied_vol_newton_root as implied_vol_newton

This also means you need to add all relevant dependencies to your BUILD rule. See, e.g., here:
https://github.com/google/tf-quant-finance/blob/873d44e3f982fe02431c4a7652f65a05a7387704/tf_quant_finance/black_scholes/BUILD#L66